returnn/tf/network.py

```diff
@@ -1347,6 +1347,7 @@ def _make_layer(layer_cls, layer_dict, map_opts=True):
         opts = nest.map_structure(_map_layer_dict_value, opts)
       opts.pop("output", None)
       opts["output"] = layer_cls.get_out_data_from_opts(**opts)
+      opts["output"] = layer_cls.fixup_out_data(**opts)
```

Comment:
My initial suggestion was to modify `out_shape` but you said that that just dropping (removing) `out_shape` should be simpler, so I assumed that you would go with that variant now.

But I don't see either of that here.

Reply:
But something like `opts["output"].batch.virtual_dims` which is the equivalent to this would not always work (e.g. without the changes to the reduce layer this would crash for the reduce layer, only after/during the call of fixup `batch` is always set correctly.

Reply:
Yes I know. That's why I agreed that you are right that it's simpler to just drop `out_shape`.